src/error/mod.rs

```diff
@@ -83,14 +82,19 @@ impl RoverError {
         self.metadata.code.clone()
     }
 
-    pub fn print(&self) -> io::Result<()> {
+    pub fn print(&self) -> RoverResult<()> {
         if let Some(RoverClientError::OperationCheckFailure {
             graph_ref: _,
             check_response,
         }) = self.error.downcast_ref::<RoverClientError>()
         {
             stdoutln!("{}", check_response.get_table())?;
         }
+        if let Some(RoverClientError::LintFailures { lint_response }) =
+            self.error.downcast_ref::<RoverClientError>()
+        {
+            stdoutln!("{}", lint_response.print_ariadne()?)?;
+        }
```

Comment:
`if let` was great here when there was only one error we were matching on, but now that there are two cases to handle, we should switch to a `match` statement so that we only call the `downcast_ref` one time, like so:

```suggestion
        match self.error.downcast_ref::<RoverClientError>() {
        Some(RoverClientError::OperationCheckFailure { graph_ref: _, check_response }) => stdoutln!("{}", check_response.get_table())?,
        Some(RoverClientError::LintFailures { lint_response }) => stdoutln!("{}", lint_response.get_ariadne()?
        }
```

Reply:
Moved to using a `match` statement